app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -93,11 +93,18 @@ def show
     param :id, :identifier, :desc => N_("content view numeric identifier"), :required => true
     param :name, String, :desc => N_("module name to restrict modules for"), :required => false
     def available_puppet_modules
-      current_uuids = @view.content_view_puppet_modules.where("uuid is NOT NULL").pluck(:uuid)
+      current_cv_puppet_modules = @view.content_view_puppet_modules.where("uuid is NOT NULL")
+      current_uuids = current_cv_puppet_modules.pluck(:uuid)
       repositories = @view.organization.library.puppet_repositories
 
       query = PuppetModule.in_repositories(repositories)
-      query = query.where(:name => params[:name]) if params[:name]
+      if params[:name]
+        query = query.where(:name => params[:name])
+        if current_uuids.present?
+          module_by_name = current_cv_puppet_modules.find_by(:name => params[:name])
+          current_uuids.delete(module_by_name.uuid) if module_by_name && module_by_name.latest_in_modules_by_author?(query)
```

Comment:
@kgaikwad, Thank you for the detailed explanation.  The new API logic appears to be what is causing the behavior that I mentioned in my other comment.

The following what I observed:
- If I select version 2.0.1,
  - with the change, available_puppet_modules is returning 2.0.1 and 1.3.0 (Unexpected extra row in the UI)
  - without the change, available_puppet_modules is returning 1.3.0 (UI behavior Ok)

- If I select version 1.3.0,
  - with the change, available_puppet_modules is returning 2.0.1 (UI behavior Ok)
  - without the change, available_puppet_modules is returning 2.0.1 (UI behavior Ok)

- If I select version 'use latest 2.0.1',
  - with the change, available_puppet_modules is returning 2.0.1 and 1.3.0 (UI behavior Ok)
  - without the change, available_puppet_modules is returning 2.0.1 and 1.3.0 (UI behavior Ok)

The ideal solution, would probably be for the 'available_puppet_modules' to take in to account the 'latest' and not return the puppet module details when it is selected.  E.g. In my case, if i select 'Use Latest (currently 2.0.1)', it would not return 2.0.1.  That said, the UI may likely also require modifications to that it can still indicate to the user the appropriate information.